tests/plugins/test_add_filesystem.py

```diff
@@ -210,45 +209,109 @@ def make_and_store_reactor_config_map(workflow, additional_koji=None):
     workflow.conf.conf = reactor_map
 
 
-@pytest.mark.skip(reason="plugin has to be reworked as it won't be importing image anymore")
 @pytest.mark.parametrize('scratch', [True, False])
-def test_add_filesystem_plugin_generated(workflow, source_dir, scratch):
-    mock_workflow(workflow, source_dir, scratch=scratch)
-    task_id = FILESYSTEM_TASK_ID
+@pytest.mark.parametrize(('dockerfile', 'expected_dockerfile'), [
+    # single-stage dockerfile with a custom image
+    (dedent('''\
+     FROM koji/image-build
+     RUN dnf install -y python-django
+     '''),
+     dedent('''\
+     FROM scratch
+     ADD {} {}
+     RUN dnf install -y python-django
+     ''')
+     ),
+    # single-stage dockerfile with a custom image and alias
+    (dedent('''\
+     FROM koji/image-build AS imported_image
+     RUN dnf install -y python-django
+     '''),
+     dedent('''\
+     FROM scratch AS imported_image
+     ADD {} {}
+     RUN dnf install -y python-django
+     ''')
+     ),
+    # multistage dockerfile with a custom image
+    (dedent('''\
+     FROM koji/image-build
+     RUN dnf install -y python-django
+     FROM test
+     RUN dnf install -b uwlg
+     '''),
+     dedent('''\
+     FROM scratch
+     ADD {} {}
+     RUN dnf install -y python-django
+     FROM test
+     RUN dnf install -b uwlg
+     ''')
+     ),
+    # multistage dockerfile with multiple custom images
+    (dedent('''
+     FROM koji/image-build
+     RUN dnf install -y python-django
+     FROM koji/image-build
+     RUN dnf install -y uwlg
+     '''),
+     dedent('''
+     FROM scratch
+     ADD {} {}
+     RUN dnf install -y python-django
+     FROM scratch
+     ADD {} {}
+     RUN dnf install -y uwlg
+     ''')
+     )
+])
+def test_add_filesystem_plugin_generated(workflow, build_dir, scratch, dockerfile,
+                                         expected_dockerfile, caplog):
+    mock_workflow(workflow, build_dir, scratch=scratch, dockerfile=dockerfile)
     mock_koji_session(scratch=scratch)
-    mock_image_build_file(source_dir)
+    mock_image_build_file(workflow)
+    image_name = 'fedora-23-1.0.x86_64.tar.gz'
+    expected_dockerfile = expected_dockerfile.replace('ADD {} {}\n',
+                                                      f'ADD {image_name} '
+                                                      '/\n')
 
     make_and_store_reactor_config_map(workflow, {'root_url': '', 'auth': {}})
+    workflow.prebuild_results[PLUGIN_RESOLVE_COMPOSES_KEY] = {'composes': []}
```

Comment:
Is this necessary? The plugin would still work if there was no resolve_composes result, right? I don't mind keeping this here, just wanted to be sure that this is not a regression

Reply:
yes it does work without it, removed it so that it's clear.